util.py

```diff
@@ -419,6 +419,109 @@ def winedll_override(dll: str, dtype: Literal['n', 'b', 'n,b', 'b,n', '']) -> No
     )
 
 
+def patch_libcuda() -> bool:
+    """Patches libcuda to work around games that crash when initializing libcuda and are using DLSS.
+
+    We will replace specific bytes in the original libcuda.so binary to increase the allowed memory allocation area.
+
+    The patched library is overwritten at every launch and is placed in .cache
+
+    Returns true if the library was patched correctly. Otherwise returns false
+    """
+    cache_dir = os.path.expanduser('~/.cache/protonfixes')
+    os.makedirs(cache_dir, exist_ok=True)
+
+    try:
+        # Use shutil.which to find ldconfig binary
+        ldconfig_path = shutil.which('ldconfig')
+        if not ldconfig_path:
+            log.warn('ldconfig not found in PATH.')
+            return False
+
+        # Use subprocess.Popen with explicit encoding handling
+        try:
+            process = subprocess.Popen(
```

Comment:
I would follow the linter's recommendation instead of working around it. Using `capture_output` simplifies things a bit and as a side effect it would also avoid the subprocess's resources (e.g., file descriptors) potentially not all being released if the subprocess's execution was interrupted.

See https://docs.astral.sh/ruff/rules/replace-stdout-stderr/

Reply:
Either would be fine:

`subprocess.run([ldconfig_path, '-p'], capture_output=True).output`

or

```
with Popen([ldconfig_path, '-p'], stdout=subprocess.PIPE, stderr=subprocess.PIPE):
  ...
```

Reply:
Solved with https://github.com/Open-Wine-Components/umu-protonfixes/pull/136/commits/17a428ec8030f57e3c26f42e9b04e1e5acf290a0